### ISSUE TITLE
Unable To Cast Type PackedScene From Resource

### ISSUE THREAD

**Issue by Supatier** [GodotNativeTools/godot-d#22]
``` 
abc@abc-pc ~/Projects/godot/godot-d-examples-one (git)-[master] % dub --compiler=dmd                                                                                                                          
Performing "debug" build using dmd for x86_64.
godot-d 0.0.1: building configuration "library"...
godot-d-examples-one ~master: building configuration "library"...
/home/abc/.dub/packages/godot-d-0.0.1/godot-d/src/godot/d/register.d(128,10): Deprecation: module `godot.d.script` member NativeScriptTemplate is not visible from module classregister
source/example_three/gameobjectfactory.d(24,34): Error: template instance `opCast!(PackedScene)` does not match template declaration opCast(T : bool)
source/example_three/gameobjectfactory.d(26,32): Error: template instance `opCast!(PackedScene)` does not match template declaration opCast(T : bool)
source/example_three/gameobjectfactory.d(28,35): Error: template instance `opCast!(PackedScene)` does not match template declaration opCast(T : bool)
dmd failed with exit code 1.

```

It's basicly `cast(PackedScene) resourceLoader.load(
                                "res://game_object/basic_object/sphere.tscn");`

Here's the [project](https://github.com/Supatier/godot-d-examples-one)

And by the way, why are godot types are structs instead of classes?

**Comment by sheepandshepherd** [comment]
Oh, this is a bug in Ref. `opCast!bool` prevents using the wrapped Resource's `opCast!PackedScene`. I'll fix it in a few minutes.

Godot classes being structs is complicated. They're wrappers around pointers to C++ classes, which allows them to override built-in behaviors of real classes. Most importantly virtual functions. Calling those functions on a real class would access the C++ vtbl, which has a different layout than the D compiler thinks, even with `extern(C++)`. Our list of functions is *only* the ones exposed to scripts, but Godot C++ classes have internal functions mixed in with them.

Making them real classes might still be possible with some hacks. We already have custom casting. `opDispatch` could be used to override how functions are called, but that would break autocompletion in IDEs and make error messages uglier. The only advantage I can see is being able to use `null` everywhere...

**Comment by sheepandshepherd** [comment]
Fix released in v0.0.2. Dub updates packages every 2 hours, so it should be there soon.
